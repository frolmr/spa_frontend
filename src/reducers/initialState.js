import messages from '../msgs'

const locale =
  (navigator.languages && navigator.languages[0])
  || navigator.language
  || navigator.userLanguage
  || 'en-US'

export default {
  posts: [],
  pages: {},
  user: {},
  intl: {
    locale: locale,
    messages: messages[locale]
  }
}
