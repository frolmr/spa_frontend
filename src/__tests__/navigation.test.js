import React from 'react';
import ReactDOM from 'react-dom';
import Navigation from '../components/navigation';
import { shallow  } from 'enzyme';
import { configure  } from 'enzyme';
import configureStore from 'redux-mock-store'
import initialState from '../reducers/initialState'
import Adapter from 'enzyme-adapter-react-16';
import { shallowWithIntl } from '../intl-enzyme-test-helper.js';

configure({ adapter: new Adapter()  });
const mockStore = configureStore();

test('home component should be rendered', () => {
  const store = mockStore(initialState);
  const wrapper = shallowWithIntl(<Navigation store={store}/>);
    expect(wrapper).toMatchSnapshot();
})
