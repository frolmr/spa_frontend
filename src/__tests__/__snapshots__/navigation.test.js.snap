// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component should be rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InjectIntl(Connect(Navigation))
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "default": Object {
            "en": Object {
              "home.welcomeMessage": "To see the posts please login first",
              "login.email": "Email",
              "login.password": "Password",
              "login.submit": "Login",
              "nav.homeLink": "Home",
              "nav.loginLink": "Login",
              "nav.profileLink": "Profile",
              "nav.searchButton": "Search",
              "postButtons.del": "Delete post",
              "postButtons.show": "Show post",
              "postForm.body": "Body",
              "postForm.name": "Name",
              "postForm.submit": "Post",
              "postForm.title": "Title",
              "postForm.url": "Image URL",
              "postpage.body": "Body",
              "postpage.title": "Title",
              "postpage.user": "User",
            },
            "ru": Object {
              "home.welcomeMessage": "Войдите, чтобы увидеть посты",
              "login.email": "Email",
              "login.password": "Пароль",
              "login.submit": "Войти",
              "nav.homeLink": "Главная",
              "nav.loginLink": "Войти",
              "nav.profileLink": "Профиль",
              "nav.searchButton": "Найти",
              "postButtons.del": "Удалить пост",
              "postButtons.show": "Показать пост",
              "postForm.body": "Текст",
              "postForm.name": "Имя",
              "postForm.submit": "Запостить",
              "postForm.title": "Заголовок",
              "postForm.url": "Ссылка на картинку",
              "postpage.body": "Текст",
              "postpage.title": "Заголовок",
              "postpage.user": "Пользователь",
            },
          },
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "default": Object {
            "en": Object {
              "home.welcomeMessage": "To see the posts please login first",
              "login.email": "Email",
              "login.password": "Password",
              "login.submit": "Login",
              "nav.homeLink": "Home",
              "nav.loginLink": "Login",
              "nav.profileLink": "Profile",
              "nav.searchButton": "Search",
              "postButtons.del": "Delete post",
              "postButtons.show": "Show post",
              "postForm.body": "Body",
              "postForm.name": "Name",
              "postForm.submit": "Post",
              "postForm.title": "Title",
              "postForm.url": "Image URL",
              "postpage.body": "Body",
              "postpage.title": "Title",
              "postpage.user": "User",
            },
            "ru": Object {
              "home.welcomeMessage": "Войдите, чтобы увидеть посты",
              "login.email": "Email",
              "login.password": "Пароль",
              "login.submit": "Войти",
              "nav.homeLink": "Главная",
              "nav.loginLink": "Войти",
              "nav.profileLink": "Профиль",
              "nav.searchButton": "Найти",
              "postButtons.del": "Удалить пост",
              "postButtons.show": "Показать пост",
              "postForm.body": "Текст",
              "postForm.name": "Имя",
              "postForm.submit": "Запостить",
              "postForm.title": "Заголовок",
              "postForm.url": "Ссылка на картинку",
              "postpage.body": "Текст",
              "postpage.title": "Заголовок",
              "postpage.user": "Пользователь",
            },
          },
        },
        "now": [Function],
        "textComponent": "span",
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "intl": Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {
            "default": Object {
              "en": Object {
                "home.welcomeMessage": "To see the posts please login first",
                "login.email": "Email",
                "login.password": "Password",
                "login.submit": "Login",
                "nav.homeLink": "Home",
                "nav.loginLink": "Login",
                "nav.profileLink": "Profile",
                "nav.searchButton": "Search",
                "postButtons.del": "Delete post",
                "postButtons.show": "Show post",
                "postForm.body": "Body",
                "postForm.name": "Name",
                "postForm.submit": "Post",
                "postForm.title": "Title",
                "postForm.url": "Image URL",
                "postpage.body": "Body",
                "postpage.title": "Title",
                "postpage.user": "User",
              },
              "ru": Object {
                "home.welcomeMessage": "Войдите, чтобы увидеть посты",
                "login.email": "Email",
                "login.password": "Пароль",
                "login.submit": "Войти",
                "nav.homeLink": "Главная",
                "nav.loginLink": "Войти",
                "nav.profileLink": "Профиль",
                "nav.searchButton": "Найти",
                "postButtons.del": "Удалить пост",
                "postButtons.show": "Показать пост",
                "postForm.body": "Текст",
                "postForm.name": "Имя",
                "postForm.submit": "Запостить",
                "postForm.title": "Заголовок",
                "postForm.url": "Ссылка на картинку",
                "postpage.body": "Текст",
                "postpage.title": "Заголовок",
                "postpage.user": "Пользователь",
              },
            },
          },
          "now": [Function],
          "textComponent": "span",
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "default": Object {
            "en": Object {
              "home.welcomeMessage": "To see the posts please login first",
              "login.email": "Email",
              "login.password": "Password",
              "login.submit": "Login",
              "nav.homeLink": "Home",
              "nav.loginLink": "Login",
              "nav.profileLink": "Profile",
              "nav.searchButton": "Search",
              "postButtons.del": "Delete post",
              "postButtons.show": "Show post",
              "postForm.body": "Body",
              "postForm.name": "Name",
              "postForm.submit": "Post",
              "postForm.title": "Title",
              "postForm.url": "Image URL",
              "postpage.body": "Body",
              "postpage.title": "Title",
              "postpage.user": "User",
            },
            "ru": Object {
              "home.welcomeMessage": "Войдите, чтобы увидеть посты",
              "login.email": "Email",
              "login.password": "Пароль",
              "login.submit": "Войти",
              "nav.homeLink": "Главная",
              "nav.loginLink": "Войти",
              "nav.profileLink": "Профиль",
              "nav.searchButton": "Найти",
              "postButtons.del": "Удалить пост",
              "postButtons.show": "Показать пост",
              "postForm.body": "Текст",
              "postForm.name": "Имя",
              "postForm.submit": "Запостить",
              "postForm.title": "Заголовок",
              "postForm.url": "Ссылка на картинку",
              "postpage.body": "Текст",
              "postpage.title": "Заголовок",
              "postpage.user": "Пользователь",
            },
          },
        },
        "now": [Function],
        "textComponent": "span",
      },
    },
  },
}
`;
